src/controllers/athlete-controller.js

```diff
@@ -0,0 +1,94 @@
+'use strict'
+
+const async = require('async')
+const boom = require('boom')
+const uuid = require('uuid/v1')
+const util = require('../util')
+const Athlete = require('../models/athlete')
+
+exports.getAthletes = (req, res, next) => {
+  Athlete.find()
+    .sort([['lastName', 'ascending'], ['firstName', 'ascending']])
+    .exec(function (err, list) {
+      if (err) { return next(err) }
+      res.status(200).json(list.map(doc => util.fromMongoDb(doc)))
+    })
+}
+
+exports.getAthlete = (req, res, next) => {
+  // todo get GHIN index from ghin.com
+  async.parallel({
+    athlete: (callback) => {
+      Athlete.findById(req.params.id)
+        .exec(callback)
+    }
+  }, function (err, results) {
+    if (err) {
+      return next(err)
+    } else if (results.athlete == null) {
+      return next(boom.notFound())
+    }
+    res.status(200).json(util.fromMongoDb(results.athlete))
+  })
+}
+
+exports.createAthlete = (req, res, next) => {
+  // todo index ghinNumber to avoid collection scan
+  Athlete.findOne({ ghinNumber: req.body.ghinNumber })
+    .exec((err, athlete) => {
+      if (err) {
+        return next(err)
+      } else if (athlete && athlete.ghinNumber && athlete.ghinNumber === req.body.ghinNumber) {
+        return next(boom.badRequest('Athlete with this GHIN number already exists'))
+      }
+      const now = Date.now()
+      const newAthlete = new Athlete(Object.assign({ _id: uuid(), createdAt: now, updatedAt: now }, req.body))
```

Comment:
Is there a reason for using uuids instead of MongoDB ids?

Reply:
I have always used uuids since I believe they are a generally accepted standard for unique ids. Also, I specifically choose v1 for uniqueness vs v4 for randomness since I am not concerned about the possibility of the loss of anonymity. In reality, I think either would be fine. I need to look more into MongoDB ids to see if they would have been acceptable.